src/transformers/modeling_lxmert.py

```diff
@@ -0,0 +1,975 @@
+            hidden_states_v=v_outputs[0][:-1],
+            attentions_v_encoder=output_v_attentions,
+            pooled_output_x_encoder=pooled_output,
+            attentions_x_encoder=output_x_attentions,
+        )
+
+
+@add_start_docstrings(
+    """Lxmert Model with a specified pre-training heads on top. """, LXMERT_START_DOCSTRING, LXMERT_INPUTS_DOCSTRING
+)
+class LxmertForPretraining(LxmertPreTrainedModel):
+    def __init__(self, config):
+        super().__init__(config)
+        # Configuration
+        self.config = config
+        self.num_answers = config.n_qa_labels
+        self.visual_loss_normalizer = config.visual_loss_normalizer
+
+        # Use of pre-training tasks
+        self.task_mask_lm = config.task_mask_lm
+        self.task_obj_predict = config.task_obj_predict
+        self.task_matched = config.task_matched
+        self.task_qa = config.task_qa
+
+        # Lxmert backbone
+        self.bert = LxmertModel(config)
+
+        # Pre-training heads
+        self.cls = LxmertPreTrainingHeads(config, self.bert.embeddings.word_embeddings.weight)
+        if self.task_obj_predict:
+            self.obj_predict_head = LxmertVisualObjHead(config)
+        if self.task_qa:
+            self.answer_head = LxmertVisualAnswerHead(config, self.num_answers)
+
+        # Weight initialization
+        self.init_weights()
+
+        # Loss functions
+        self.loss_fcts = {
+            "l2": SmoothL1Loss(reduction="none"),
+            "visn_ce": CrossEntropyLoss(ignore_index=-1, reduction="none"),
+            "ce": CrossEntropyLoss(ignore_index=-1),
+        }
+
+        visual_losses = {}
+        if config.visual_obj_loss:
+            visual_losses["obj"] = {"shape": (-1,), "num": config.n_object_labels, "loss": "visn_ce"}
+        if config.visual_attr_loss:
+            visual_losses["attr"] = {"shape": (-1,), "num": config.n_attr_labels, "loss": "visn_ce"}
+        if config.visual_obj_loss:
+            visual_losses["feat"] = {
+                "shape": (-1, config.visual_feat_dim),
+                "num": config.visual_feat_dim,
+                "loss": "l2",
+            }
+        self.visual_losses = visual_losses
+
+    def forward(
+        self,
+        input_ids,
+        visual_feats,
+        token_type_ids=None,
+        attention_mask=None,
+        masked_lm_labels=None,
+        obj_labels=None,
+        matched_label=None,
+        ans=None,
+        output_v_attentions=False,
+        output_l_attentions=False,
+        output_x_attentions=False,
+        return_dict=False,
+        **kwargs
+    ):
+
+        (visual_feats, pos) = visual_feats
+
+        r"""
+        masked_lm_labels (``torch.LongTensor`` of shape ``(batch_size, sequence_length)``, `optional`, defaults to :obj:`None`):
+            Labels for computing the masked language modeling loss.
+            Indices should be in ``[-100, 0, ..., config.vocab_size]`` (see ``input_ids`` docstring)
```

Comment:
If you want to keep the `ignore_index=-1` in the cross entropy losses, this docstring and the ones below should be updated to reflect that.